sw/device/lib/crypto/aes_gcm/aes_gcm.c

```diff
@@ -354,3 +373,93 @@ aes_error_t aes_gcm_ghash(const aes_block_t *hash_subkey,
 
   return kAesOk;
 }
+
+/**
+ * One-shot version of the AES API for a single block, for convenience.
+ */
+static aes_error_t aes_single_block(const aes_params_t params,
+                                    const aes_block_t *input,
+                                    aes_block_t *output) {
+  aes_error_t err = aes_begin(params);
+  if (err != kAesOk) {
+    return err;
+  }
+  err = aes_update(output, input);
+  if (err != kAesOk) {
+    return err;
+  }
+  err = aes_end();
+  if (err != kAesOk) {
+    return err;
+  }
+  return kAesOk;
+}
+
+/**
+ * Implements the GCTR function as specified in SP800-38D, section 6.5.
+ *
+ * The block cipher is fixed to AES. Note that the GCTR function is a modified
+ * version of the AES-CTR mode of encryption.
+ *
+ * @param aes_ctr_params AES parameters for CTR encryption with IV set to the
```

Comment:
I think we can but I'm not sure we do at the moment; I couldn't find any from a quick search. But I always include the Doxygen-style annotations since they're [recommended by the style guide](https://docs.opentitan.org/doc/rm/c_cpp_coding_style/#public-function-api-documentation) in any case.

Reply:
Oh neat, I found `util/build_docs.py --preview` can build and host docs locally. I think they're hosted here? https://docs.opentitan.org/sw/apis/index.html

Reply:
Yep, I looked there but didn't find much that looked like it was being created automatically; at least, if so, it's limited to a few files.

Reply:
We can probably add some more details to the style guides on how to integrate the header into the doxygen docs (we may be missing some markup blocks). I think in most cases we just document the API level though (.h). 

Regardless, these doc strings are useful for IDEs et al.